go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/cpalsulich/simple-chat-server
+
+go 1.12
```

Comment:
Can upgrade to 1.13, which was released a couple weeks ago. The main feature relevant to this PR is being able to use error wrapping with `%w` in `fmt.Errorf` calls. Essentially, do `fmt.Errorf("something failed: %w", err)` instead of `%v`.

https://golang.org/pkg/errors/